core/src/main/java/com/netflix/conductor/core/execution/WorkflowExecutor.java

```diff
@@ -765,8 +765,9 @@ public void terminateWorkflow(Workflow workflow, String reason, String failureWo
 		// Send to atlas
 		Monitors.recordWorkflowTermination(workflow.getWorkflowType(), workflow.getStatus());
 
-		if (lazyDecider && workflow.getStatus() == WorkflowStatus.RESET && StringUtils.isNotEmpty(workflow.getParentWorkflowId())) {
-			wakeUpSweeper(workflow.getParentWorkflowId());
+		// ONECOND-794: https://jira.d3nw.com/browse/ONECOND-794
+		if (lazyDecider && workflow.getStatus() == WorkflowStatus.RESET && workflow.isSubWorkflow()) {
+			rewind(workflow.getWorkflowId(), workflow.getCorrelationId());
```

Comment:
This code introduces a new behaviour which is to always restart regardless of the sub-workflow configuration (restartCount, standbyOnFail, restartOnFail). See SubWorkflowParams.java